src/Model/Union.php

```diff
@@ -0,0 +1,464 @@
+                        unset($fields[$k]);
+                    }
+                }
+                $subquery = $this->getSubQuery($fields);
+                $query = parent::action($mode, $args)->reset('table')->table($subquery);
+
+                if (isset($this->group)) {
+                    $query->group($this->group);
+                }
+                $this->hook(self::HOOK_AFTER_SELECT, [$query]);
+
+                return $query;
+            case 'count':
+                $subquery = $this->getSubAction('count', ['alias' => 'cnt']);
+
+                $mode = 'fx';
+                $args = ['sum', $this->expr('{}', ['cnt'])];
+
+                break;
+            case 'field':
+                if (!isset($args[0])) {
+                    throw (new Exception('This action requires one argument with field name'))
+                        ->addMoreInfo('mode', $mode);
+                }
+
+                if (!is_string($args[0])) {
+                    throw (new Exception('action "field" only support string fields'))
+                        ->addMoreInfo('field', $args[0]);
+                }
+
+                $subquery = $this->getSubQuery([$args[0]]);
+
+                break;
+            case 'fx':
+                $subquery = $this->getSubAction('fx', [$args[0], $args[1], 'alias' => 'val']);
+
+                $args = [$args[0], $this->expr('{}', ['val'])];
+
+                break;
+            default:
+                throw (new Exception('Union model does not support this action'))
+                    ->addMoreInfo('mode', $mode);
+        }
+
+        // Substitute FROM table with our subquery expression
+        return parent::action($mode, $args)->reset('table')->table($subquery);
+    }
+
+    /**
+     * Export model.
+     *
+     * @param array|null $fields        Names of fields to export
+     * @param string     $key_field     Optional name of field which value we will use as array key
+     * @param bool       $typecast_data Should we typecast exported data
+     */
+    public function export($fields = null, $key_field = null, $typecast_data = true): array
+    {
+        if ($fields) {
+            $this->onlyFields($fields);
+        }
+
+        $data = [];
+        foreach ($this->getIterator() as $row) {
+            $data[] = $row->get();
+        }
+
+        return $data;
+    }
+
+    /**
+     * Adds nested model in union.
+     *
+     * @param string|Model $class    model
+     * @param array        $fieldMap Array of field mapping
+     */
+    public function addNestedModel($class, array $fieldMap = []): Model
+    {
+        $nestedModel = $this->persistence->add($class);
+
+        $this->union[] = [$nestedModel, $fieldMap];
+
+        return $nestedModel;
+    }
+
+    /**
+     * Specify a single field or array of fields.
+     *
+     * @param string|array $group
+     *
+     * @return $this
+     */
+    public function groupBy($group, array $aggregate = [])
```

Comment:
probably not, union is about union only

Reply:
Yes. There should not be grouping in union model anymore.
Grouping is group model and union should be only about unions.

If you need both, then you should be able to easily use both model classes in conjunction.

Reply:
> If you need both, then you should be able to easily use both model classes in conjunction.

exactly, this is not possible when 2 classes are extended